api/app/Http/Controllers/UserController.php

```diff
@@ -2,15 +2,33 @@
 
 namespace App\Http\Controllers;
 
+use App\Models\College;
+use App\Models\Course;
+use App\Models\Department;
 use App\Models\User;
+use Illuminate\Contracts\Database\Eloquent\Builder;
 use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
 
 class UserController extends Controller
 {
-    //
+    public function getAuthUser()
+    {
+        $user = Auth::user();
+        $id = Auth::id();
+
+        if ($user->roleUser->role_id === 2) {
+            $user = User::with('university.colleges.departments.courses')->find($id);
+            return response()->json($user);
+        } else {
+            $user = User::with('department.college.university', 'courseUsers.course')->find($id);
+            return response()->json($user);
+        }
+    }
```

Comment:
Put return statement at the end of the condition
```suggestion
        }
        return response()->json($user);
    }
```

Reply:
Comment applied.